docs/intro-react-native-components.md

```diff
@@ -0,0 +1,100 @@
+
+<div class="toggler">
+  <ul role="tablist" id="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTab('syntax', 'functional')">
+      Functional Syntax
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTab('syntax', 'classical')">
+      Class Syntax
+    </li>
+  </ul>
+</div>
+
+<block class="functional" />
+
+```SnackPlayer name=Hello%20World
+import React from 'react';
+import { View, Text, Image, ScrollView, TextInput } from 'react-native';
+
+export default function App() {
+    return (
+      <ScrollView>
+        <Text>ScrollViews scroll!</Text>
+        <View>
+          <Text>Some text</Text>
+          <Image source="https://placekitten.com/g/200/300" style={{width: 200, height: 200}}/>
+        </View>
+        <TextInput
+          style={{ height: 40, borderColor: 'gray', borderWidth: 1 }}
+          defaultValue="You can type in me"
+        />
+      </ScrollView>
+    );
+}
+```
+
+<block class="classical" />
+
+```SnackPlayer name=Hello%20World
+import React, { Component } from 'react';
+import { View, Text, Image, ScrollView, TextInput } from 'react-native';
+
+export default class App extends Component {
+  render() {
+    return (
+      <ScrollView>
+        <Text>ScrollViews scroll!</Text>
+        <View>
+          <Text>Some text</Text>
+          <Image source="https://placekitten.com/g/200/300" style={{width: 200, height: 200}}/>
+        </View>
+        <TextInput
+          style={{ height: 40, borderColor: 'gray', borderWidth: 1 }}
+          defaultValue="You can type in me"
+        />
+      </ScrollView>
+    );
+  }
+}
+
+```
+
+<block class="classical functional" />
+
+## Make your own Native Components
+
+As you continue developing with React Native, you may want to use a kind of view on Android or iOS that isn’t in the Core Components. For this reason, React Native lets you to build your own Native Components for [Android](native-components-android.md) and [iOS](native-components-ios.md) to suit your app's unique needs. We also have a thriving ecosystem of these community-contributed components and modules. Check [Native Directory](https://www.native.directory/) to find what the community has been creating.
+
+---
+
+Because React Native uses the same API structure as React components, you’ll need to understand React component APIs to get started. The [next section](intro-react) makes for a quick introduction or refresher on the topic. However, if you’re already familiar with React, feel free to [skip ahead](handling-text-input).
+
+<img src="/react-native/docs/assets/react-native-components.png" srcset="/react-native/docs/assets/react-native-components2x.png 2x" width="1000" alt="A diagram showing React Native's Core Components are a subset of React Components that ship with React Native.">
```

Comment:
Is this diagram intentionally on the bottom after the text asking me to navigate away from this page?

Reply:
Diagram nit: I find most of these "s" at the end a bit hard to read as a non-native speaker. (Cause I don't draw it like this.) Overall I would prefer a tiny bit more polished diagram.

Reply:
@cpojer It's at the bottom as a sort of visual endcap.

Does it add value? I found it useful working with Luna and Kirsty for explaining how all the components relate to each other.

I'm happy to polish it (originally intended to, but then I realized I could spend 5 hours making a diagram no one found useful and wanted to see what reviewers thought first! I am sneaky like this.)